content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,608 @@
+presence of the same language as in the training dataset (i.e. English), we obtain poorer results than before. That situation
+is mostly due to the postal code format is not similar. For the US, it is 5 digits, and for the UK it is similar to Canada, but it is not always a letter followed by a number, and it is not always 6 characters. It's *normal* for a model to
+have difficulty is that kind of new pattern, but he has still achieved good results.
+
+| Model (Country) | LSTM one layer | Bi-LSTM two layers |
+|:---------------:|:--------------:|:------------------:|
+|    Loss (US)    |     0.6176     |       **0.3078**   |
+|  Accuracy (US)  |     84.7396    |       **91.8220**  |
+|    Loss (UK)    |     0.4368     |       **0.1571**   |
+|  Accuracy (UK)  |     86.2543    |       **95.6840**  |
+
+
+##### The Second and Third Test
+
+Now let's tests for Russia and Mexico.
+
+```python
+ru_loader = DataLoader(ru_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(ru_loader)
+exp_bi_lstm.test(ru_loader)
+
+mx_loader = DataLoader(mx_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(mx_loader)
+exp_bi_lstm.test(mx_loader)
+```
+
+The next table presents the results of both the model for both the country tested. We first see that the first test 
+(RU) gives poorer results than Mexico, even if the second one is a different structure and language. 
+This situation could be explained by both languages' language roots; Spanish is closer to French than Russia. 
+But an interesting thing is that even in a *difficult* annotation context, both the model perform relatively well. 
+It means that our models seem to have really learned the *logic* of an address sequence. It could also mean that if 
+we train our model longer, maybe we could improve our results. But, other improvements will be discussed in the next summary section.
+
+| Model (Country) | LSTM one layer | Bi-LSTM two layers |
+|:---------------:|:--------------:|:------------------:|
+|    Loss (RU)    |   **2.5181**   |       4.6118       |
+|  Accuracy (RU)  |   **48.9820**  |       47.3185      |
+|    Loss (MX)    |     2.6786     |     **1.7147**     |
+|  Accuracy (MX)  |     50.2013    |     **63.5317**    |
+
+### Summary
+In summary, we found that using a Bi-LSTM with two layers seems to perform better on countries' addresses not seen during training. Still, the results are not as good as those of Canada (training dataset). A solution to this problem could be to train a model using all the
+possible data in the world. This approach was used by [Libpostal](https://github.com/openvenues/libpostal) which trained a 
+CRF over an impressive near `100` million address (yes, **100 million**). The data is publicly available if you want to explore this avenue next.
+
+We also explored that the language has a negative impact on the results since we use monolingual word embeddings (i.e. French), 
```

Comment:
Ce serait bien de clarifier cette phrase.
```suggestion
We also explored the idea that the language has a negative impact on the results, since we use monolingual word embeddings (i.e. French), 
```